src/controller/userController.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const userServise = require('../services/users.js');
+
+const post = (req, res) => {
+  const { name } = req.body;
+
+  if (!name || typeof name !== 'string') {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newUser = {
+    id: Math.floor(Math.random()),
+    name,
+  };
+
+  userServise.add(newUser);
+
+  res.statusCode = 201;
+  res.send(newUser);
+};
+
+const get = (req, res) => {
+  const users = userServise.getAll();
+
+  res.statusCode = 200;
+
+  if (users.length === 0) {
+    res.send([]);
+  }
+
+  res.send(users);
+};
+
+const getId = (req, res) => {
+  const { id } = req.params;
+  const foundUser = userServise.findById(+id);
+
+  if (!foundUser) {
+    res.sendStatus(404);
+
+    return;
+  };
+
+  if (typeof +id !== 'number') {
+    res.sendStatus(422);
+
+    return;
+  };
+
+  res.send(foundUser);
+};
+
+const patch = (req, res) => {
+  const { id } = req.params;
+  const foundUser = userServise.findById(+id);
+
+  if (!foundUser) {
+    res.sendStatus(404);
+
+    return;
+  };
+
+  if (typeof +id !== 'number') {
+    res.sendStatus(422);
```

Comment:
I think `400` would be better in this case. Because 422 means:
 **the server understands the content type of the request entity**, and the syntax of the request entity **is correct**, but it was unable to process the contained instructions
 
 What do you think about it?